sway/desktop/output.c

```diff
@@ -268,6 +291,22 @@ static int output_repaint_timer_handler(void *data) {
 			wlr_output_state_set_gamma_lut(&pending, 0, NULL, NULL, NULL);
 		}
 	}
+	
+	if (output->can_tear) {
+		pending.tearing_page_flip = true;
+		
+		if (!wlr_output_test_state(output->wlr_output, &pending)) {
+			sway_log(SWAY_ERROR, "Output test failed on '%s', retrying without tearing page-flip",
+				output->wlr_output->name);
+				
+			pending.tearing_page_flip = false;
+			
+			if (!wlr_output_test_state(output->wlr_output, &pending)) {
+				wlr_output_state_finish(&pending);
+				return 0;
+			}
```

Comment:
This one can be removed: we're doing a real commit right after, and the real commit implicitly performs a test before being applied.

Reply:
Removed.